src/Core/3DTiles/C3DTBatchTableHierarchyExtension.js

```diff
@@ -69,46 +73,36 @@ class BatchTableHierarchyExtension extends $3dTilesAbstractExtension {
      * @returns {Object} - displayable information relative to this extension
      * and for the feature with id=featureId and for its parents
      */
-    getPickingInfo(featureId) {
+    getInfoById(featureId) {
         const instanceProperties = {};
         // get feature class name
         const instanceClassId = this.instancesIdxs[featureId].classId;
         const featureClass = this.classes[instanceClassId].name;
         // get feature properties and values
         const instanceIdx = this.instancesIdxs[featureId].instanceIdx;
-        Object.keys(this.classes[instanceClassId].instances)
-            .forEach((property) => {
-                instanceProperties[property] =
-                    this.classes[instanceClassId].instances[property][instanceIdx];
-            });
+
+        const properties = Object.keys(this.classes[instanceClassId].instances);
+        for (let i = 0; i < properties.length; i++) {
+            const property = properties[i];
+            instanceProperties[property] =
+                this.classes[instanceClassId].instances[property][instanceIdx];
+        }
         // create return object: className: {featureProperties and values}
         const pickingInfo = {};
         pickingInfo[featureClass] = instanceProperties;
         // If this feature has parent(s), recurse on them
         if (this.inverseHierarchy && this.inverseHierarchy[featureId]) {
-            this.inverseHierarchy[featureId].forEach(
-                parentId => Object.assign(pickingInfo,
-                    this.getPickingInfo(parentId)));
+            for (let i = 0; i < this.inverseHierarchy[featureId].length; i++) {
+                const parentID = this.inverseHierarchy[featureId][i];
+                Object.assign(pickingInfo, this.getInfoById(parentID));
+            }
         }
         return pickingInfo;
     }
 }
 
-/**
- * @module BatchTableHierarchyExtensionParser
- */
-export default {
-    /**
-     * Parses a
-     * [3DTILES_batch_table_hierarchy extension]{@link https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/master/extensions/3DTILES_batch_table_hierarchy}
-     * and returns a Promise that resolves with a BatchTableHierarchyExtension
-     * object.
-     * @param {Object} json - json holding the extension
-     * @return {Promise} - a promise that resolves with a
-     *     BatchTableHierarchyExtension object.
-     */
-    parse(json) {
-        return Promise.resolve(new BatchTableHierarchyExtension(json));
-    },
-};
+function C3DT_BTHExtension_BatchTable_parser(json) {
```

Comment:
I can register an instance of the class and move the content of the constructor to a parse method of the class; would you prefer that ?

Reply:
I meant : could register `C3DT_BTHExtension_BatchTable` instead of parser:

```js
extensions.registerExtension('3DTILES_batch_table_hierarchy',
        { batchtable: C3DT_BTHExtension_BatchTable });
```

Reply:
No this is not possible, I can only register a class instance or a function. Maybe registering a class instance is clearer than registering `C3DT_BTHExtension_BatchTable_parser`.

Reply:
I've tried this and it works.

The registering:
```js
extensions.registerExtension("3DTILES_batch_table_hierarchy",
    { [itowns.C3DTilesTypes.batchtable]:  itowns.C3DT_BTHExtension_BatchTable });
```

The instancing in `C3DTExtensions#parseExtensions`
```js
parsedExtensions[extName] = new (this.getExtension(extName, type))(extensionsJSON[extName]);
```

Reply:
Nice, thanks ! I must have made a mistake when I tried it. It is simpler this way.